R/tree-sequences.R

```diff
@@ -323,27 +348,25 @@ ts_simplify <- function(ts, simplify_to = NULL, spatial = TRUE, keep_input_roots
     return(ts)
   }
 
-  if (is.null(simplify_to) && backend == "SLiM")
+  if (is.null(simplify_to)) {
     samples <- dplyr::filter(data, sampled)$node_id
-  else if (is.character(simplify_to)) {
+  } else if (is.character(simplify_to)) {
+    if (is.null(model))
```

Comment:
seems like you need a better way to check "is this slendr-generated" than `!is.null(model)`

Reply:
I'm utterly confused by GitHub's PR commenting interface (first time using it!) so I'm just acknowledging you pointing this out. I will check what this looks like in the current version of the code and fix things as needed. Thanks!

Reply:
I've just noticed that you're doing `is.null(model)` a lot, and I think it's to mean "is this slendr-generated"; it seems like a good thing to abstract out?

Reply:
Yeah, it's not very pretty, is it? I was mostly sprinkling them here and there to see what slendr-specific code needs to be switched of or tweaked to work even without the slendr metadata (names of sampled individuals, coordinate reference systems, that sort of thing).

I will see if creating a different attribute type I'm attaching to a tree sequence R object could be used for this. Right now I have an attribute `source = <msprime|SLiM>` but based on the discussion above this is not actually a very meaningful bit of info for what's going on during the loading/processing.

Reply:
I have now refactored the code so that all those `is.null(model)` calls check to test if the currently processed/analyzed tree sequence are more readable. In a number of places I was able to join multiple of those conditions in a single place so the code is cleaner now too. The old version was quite ugly and it was all too clear I just injected them in multiple places as special case triggers.